src/mruudecode.c

```diff
@@ -59,8 +89,725 @@ char* mruudecode_do(const char* text, char** ret_binary, size_t* ret_binary_byte
 		goto cleanup; // bad parameters
 	}
 
+    /**
+     *  new code from cs which works with uuencoded parts
+     */
+    char* uustartpos    = NULL;
+    char* ret_msg_text  = NULL; // NULL = no uuencoded parts are found (standard case)
+    
+    
+    if (++uu_part_loop_cnt > MAX_DECODING_LOOPS){
+        // fence against infinite loops
+        goto cleanup;
+    }
+
+    if (line_end_pattern == NULL){
+        // we are in the first loop -> init
+        
+        line_end_pattern = mr_detect_line_end(text);
+        if(line_end_pattern == NULL){
+            #ifdef __DEBUG__
+            printf("Error - Line End Pattern not detected - stopping here !\n\n");
+            #endif
+            goto cleanup;
+        }
+        #ifdef __DEBUG__
+        printf("Line End = '%s'\n\n", line_end_pattern);
+        #endif
+    }
+    
+    uustartpos = mr_find_uuencoded_part(text);
+    if (!uustartpos){
+        #ifdef __DEBUG__
+        puts("no or no further uu_part found\n\n");
+        puts("printing now message after all mr_handle_uuencoded_part(s)\n\n");
+        gets(in);
+        puts(text);
+        #endif
+
+        // no or no further uu_parts found
+        goto cleanup;
+    }
+
+    #ifdef __DEBUG__
+    printf("uu_part found at pos: %.50s [......]\n\n", uustartpos);
+    #endif
+
+    ret_msg_text = mr_handle_uuencoded_part (text, uustartpos, ret_binary, ret_binary_bytes, ret_filename);
+    
+    #ifdef __DEBUG__
+      puts("printing now message after mr_handle_uuencoded_part\n\n");
+      gets(in);
+      puts("go ....\n\n");
+    #endif
+
+    return ret_msg_text;
+    
+
 cleanup:
+    // reset globals
+    line_end_pattern = NULL;
+    uu_part_loop_cnt = 0;
+
 	return NULL;
 }
 
 
+
+/**
+ *  From here on new code from  Christian Schneider <schneider17@gmx.de>  follows
+ *
+ *  This is version 2 of the mruudecode.c which uses the api which is used in
+ *  mrmimeparser.c or the original project.
+ */ 
+
+
+/**
+ *  Helper functions
+ */
+
+char* mr_print_hex(char* s){
+    /**
+     * make hex representation of a string
+     * 
+     * Author: Christian Schneider <schneider17@gmx.de>
```

Comment:
we should not add an "author" line to function bodies. this may be interpreted as if this claims additional/other licenses as stated in the header or in the LICENSE file and it is important for the whole core to be available under the same license from the header. moreover, this may become confusing, if the code is edited by someone else, is moved around etc.
for easy contacting, i think the git log is sufficient. 
moreover, feel free to add your name to the contributors list at https://github.com/deltachat/deltachat-pages/blob/master/_includes/contributors.md (see https://github.com/deltachat/deltachat-pages/issues/71 )

Reply:
Good proposal. Done :-)